backend/routes/orders.js

```diff
@@ -184,4 +184,29 @@ module.exports = function (router) {
     },
     makeOffer
   )
+
+  /**
+   * To update the payment state of an order
+   *
+   * @param {String} paymentCode the custom ID of the external payment
+   * @param {enums.OrderPaymentStatuses} state new payment state to set
+   */
+  router.put(
+    '/orders/:orderId/payment-state',
+    authSellerAndShop,
+    findOrder,
+    async (req, res) => {
+      const { state } = req.body
+
+      const order = req.order
+
+      if (!order) {
```

Comment:
I don't think it's necessary since the findOrder middleware returns a 404 in case the order is not found.